sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala

```diff
@@ -51,223 +51,499 @@ trait TPCDSBase extends SharedSparkSession {
+        |`c_first_name` CHAR(20),
+        |`c_last_name` CHAR(30),
+        |`c_preferred_cust_flag` CHAR(1),
+        |`c_birth_day` INT,
+        |`c_birth_month` INT,
+        |`c_birth_year` INT,
+        |`c_birth_country` VARCHAR(20),
+        |`c_login` CHAR(13),
+        |`c_email_address` CHAR(50),
+        |`c_last_review_date` INT
       """.stripMargin,
     "customer_address" ->
       """
-        |`ca_address_sk` INT, `ca_address_id` STRING, `ca_street_number` STRING,
-        |`ca_street_name` STRING, `ca_street_type` STRING, `ca_suite_number` STRING,
-        |`ca_city` STRING, `ca_county` STRING, `ca_state` STRING, `ca_zip` STRING,
-        |`ca_country` STRING, `ca_gmt_offset` DECIMAL(5,2), `ca_location_type` STRING
+        |`ca_address_sk` INT,
+        |`ca_address_id` CHAR(16),
+        |`ca_street_number` CHAR(10),
+        |`ca_street_name` VARCHAR(60),
+        |`ca_street_type` CHAR(15),
+        |`ca_suite_number` CHAR(10),
+        |`ca_city` VARCHAR(60),
+        |`ca_county` VARCHAR(30),
+        |`ca_state` CHAR(2),
+        |`ca_zip` CHAR(10),
+        |`ca_country` VARCHAR(20),
+        |`ca_gmt_offset` DECIMAL(5,2),
+        |`ca_location_type` CHAR(20)
       """.stripMargin,
     "customer_demographics" ->
       """
-        |`cd_demo_sk` INT, `cd_gender` STRING, `cd_marital_status` STRING,
-        |`cd_education_status` STRING, `cd_purchase_estimate` INT, `cd_credit_rating` STRING,
-        |`cd_dep_count` INT, `cd_dep_employed_count` INT, `cd_dep_college_count` INT
+        |`cd_demo_sk` INT,
+        |`cd_gender` CHAR(1),
+        |`cd_marital_status` CHAR(1),
+        |`cd_education_status` CHAR(20),
+        |`cd_purchase_estimate` INT,
+        |`cd_credit_rating` CHAR(10),
+        |`cd_dep_count` INT,
+        |`cd_dep_employed_count` INT,
+        |`cd_dep_college_count` INT
       """.stripMargin,
     "date_dim" ->
       """
-        |`d_date_sk` INT, `d_date_id` STRING, `d_date` DATE,
-        |`d_month_seq` INT, `d_week_seq` INT, `d_quarter_seq` INT, `d_year` INT, `d_dow` INT,
-        |`d_moy` INT, `d_dom` INT, `d_qoy` INT, `d_fy_year` INT, `d_fy_quarter_seq` INT,
-        |`d_fy_week_seq` INT, `d_day_name` STRING, `d_quarter_name` STRING, `d_holiday` STRING,
-        |`d_weekend` STRING, `d_following_holiday` STRING, `d_first_dom` INT, `d_last_dom` INT,
-        |`d_same_day_ly` INT, `d_same_day_lq` INT, `d_current_day` STRING,
-        |`d_current_week` STRING, `d_current_month` STRING, `d_current_quarter` STRING,
-        |`d_current_year` STRING
+        |`d_date_sk` INT,
+        |`d_date_id` CHAR(16),
+        |`d_date` DATE,
+        |`d_month_seq` INT,
+        |`d_week_seq` INT,
+        |`d_quarter_seq` INT,
+        |`d_year` INT,
+        |`d_dow` INT,
+        |`d_moy` INT,
+        |`d_dom` INT,
+        |`d_qoy` INT,
+        |`d_fy_year` INT,
+        |`d_fy_quarter_seq` INT,
+        |`d_fy_week_seq` INT,
+        |`d_day_name` CHAR(9),
+        |`d_quarter_name` CHAR(1),
```

Comment:
This should be `char(6)` (see p28 in http://www.tpc.org/tpc_documents_current_versions/pdf/tpc-ds_v2.9.0.pdf)

Reply:
https://github.com/apache/spark/pull/31886/files#r599317278

Reply:
I'll fix this in myPR #31886.

Reply:
nice catch!

Reply:
Can we do that separately becuase #31886 is about adding a new GitHub Action job?

Reply:
Yea, sure, @dongjoon-hyun ~

Reply:
https://github.com/apache/spark/pull/31943